tests/testthat/test-select.R

```diff
@@ -0,0 +1,292 @@
+    expect_equivalent(attr(out, "span")$rows,
+                      attr(qtable.3D.nominal.multi.multistat,
+                           "span")$rows[(n-4):n, , drop = FALSE])
+})
+
+test_that("last and first date selection mode",
+{
+    dates <- seq(as.Date("2000/1/15"), as.Date("2001/01/15"), length.out = 14)
+    x <- matrix(seq_len(length(dates)*5), length(dates), 5)
+    colnames(x) <- LETTERS[1:ncol(x)]
+    date.fmt <- "%Y/%m/%d"
+    rownames(x) <- format(dates, date.fmt)
+    out <- SelectFromTable(x, row.selection.mode = "first date-time periods",
+                           row.selections = 1, unit = "Quarter", calendar = TRUE)
+    expect_equal(dim(out), c(3, ncol(x)))
+    expect_equivalent(out, First(x, keep = 1, unit = "Quarter"))
+    expect_equal(rownames(out), format(dates[1:3], date.fmt))
+
+    out <- SelectFromTable(x, row.selection.mode = "first date-time periods",
+                           row.selections = 1, unit = "Quarter", calendar = FALSE)
+    expect_equal(dim(out), c(4, ncol(x)))
+    expect_equal(out, First(x, keep = 1, unit = "Quarter", calendar = FALSE))
+
+    out <- SelectFromTable(x, row.selection.mode = "first date-time periods",
+                           row.selections = 1, unit = "Year", calendar = FALSE)
+    expect_equal(dim(out), c(nrow(x) - 1, ncol(x)))
+
+
+    out <- SelectFromTable(x, row.selection.mode = "last date-time periods",
+                           row.selections = 1, unit = "Year", calendar = FALSE)
+    expect_equal(out, Last(x, keep = 1, unit = "Year", calendar = FALSE))
+    out <- SelectFromTable(x, row.selection.mode = "last date-time periods",
+                           row.selections = 1, unit = "Year", calendar = TRUE)
+    expect_equal(dim(out), c(1, ncol(x)))
+
+    x <- t(x)
+    out <- SelectFromTable(x, column.selection.mode = "first date-time periods",
+                           column.selections = 3, unit = "Month")
+    expect_equal(dim(out), c(nrow(x), 3))
+    out <- SelectFromTable(x, column.selection.mode = "first date-time periods",
+                           column.selections = 3, unit = "Month", calendar = FALSE)
+    expect_equal(dim(out), c(nrow(x), 4))
+
+    out <- SelectFromTable(x, column.selection.mode = "last date-time periods",
+                           column.selections = 5, unit = "Week")
+    expect_equal(dim(out), c(nrow(x), 2))
+    expect_equal(out, Last(out, keep = 5, unit = "Week"))
+
+    row.keep <- c(2, 4)
+    out <- SelectFromTable(x, row.selection.mode = "vector",
+                           row.selections = LETTERS[row.keep],
+                           column.selection.mode = "last date-time periods",
+                           column.selections = 30, unit = "Day", calendar = FALSE)
+    expect_equal(dim(out), c(2, 2))
+    expect_equivalent(out, Last(x, keep = 30, unit = "Day",
+                                calendar = FALSE)[row.keep, ])
+})
+
+test_that("range selection mode",
```

Comment:
Need to add tests for bad inputs, e.g. incorrect format, out of range, out of order

Reply:
Also test what happens with duplicate selections

Reply:
I've added the tests. I've also added a call to `unique()` so duplicate selections are now ignored.

Reply:
Could you commit the changes as I don't see them yet